Hardware/Nvidia/NVML.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace OpenHardwareMonitor.Hardware.Nvidia {
+    internal class NVML {
+        internal bool Initialised { get; }
+
+        private const string WindowsDllName = "nvml";
+        private const string LinuxDllName = "nvidia-ml";
+
+        internal NVML() {
+            if (Software.OperatingSystem.IsLinux) {
+                try {
+                    Initialised = (LinuxNvmlInit() == NvmlReturn.Success);
+                }
+                catch (DllNotFoundException) { return; }
+                catch (EntryPointNotFoundException) {
+                    try {
+                        Initialised = (LinuxNvmlInitLegacy() == NvmlReturn.Success);
+                    }
+                    catch (EntryPointNotFoundException) { return; }
+                }
+            }
+            else if (IsNvmlCompatibleWindowsVersion()) {
+                var programFilesDirectory = Environment.ExpandEnvironmentVariables("%ProgramW6432%");
+                var dllPath = Path.Combine(programFilesDirectory, @"NVIDIA Corporation\NVSMI\");
+                SetDllDirectory(dllPath);
```

Comment:
I don't think this should be used, as it will affect all other DLL calls as well (ADL from AMD uses DllImport as well for example). I think AddDllDirectory can be used instead (don't have NVIDIA to test at the moment).
Also, is it guaranteed to be in this folder? What if you install the drivers to another location?

Reply:
Hi @Phyxion, I don't believe you can change the NVSMI install folder (although a user may move the dll after install)